format-xls/src/main/java/io/cdap/plugin/format/xls/input/XlsInputFormatProvider.java

```diff
@@ -0,0 +1,192 @@
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.format.xls.input;
+
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.annotation.Plugin;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.plugin.PluginClass;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.cdap.etl.api.validation.FormatContext;
+import io.cdap.cdap.etl.api.validation.InputFile;
+import io.cdap.cdap.etl.api.validation.InputFiles;
+import io.cdap.cdap.etl.api.validation.ValidatingInputFormat;
+import io.cdap.plugin.format.input.PathTrackingConfig;
+import io.cdap.plugin.format.input.PathTrackingInputFormatProvider;
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.DataFormatter;
+import org.apache.poi.ss.usermodel.FormulaEvaluator;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.Workbook;
+import org.apache.poi.ss.usermodel.WorkbookFactory;
+import org.apache.poi.ss.util.CellReference;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+
+/**
+ * Reads XLS(X) into StructuredRecords.
+ */
+@Plugin(type = ValidatingInputFormat.PLUGIN_TYPE)
+@Name(XlsInputFormatProvider.NAME)
+@Description(XlsInputFormatProvider.DESC)
+public class XlsInputFormatProvider extends PathTrackingInputFormatProvider<XlsInputFormatConfig> {
+  static final String NAME = "xls";
+  static final String DESC = "Plugin for reading files in xls(x) format.";
+  public static final PluginClass PLUGIN_CLASS = PluginClass.builder()
+          .setType(ValidatingInputFormat.PLUGIN_TYPE)
+          .setName(NAME)
+          .setDescription(DESC)
+          .setClassName(XlsInputFormatProvider.class.getName())
+          .setConfigFieldName("conf")
+          .setProperties(XlsInputFormatConfig.XLS_FIELDS)
+          .build();
+  private final XlsInputFormatConfig conf;
+
+  public XlsInputFormatProvider(XlsInputFormatConfig conf) {
+    super(conf);
+    this.conf = conf;
+  }
+
+  @Override
+  public String getInputFormatClassName() {
+    return XlsInputFormat.class.getName();
+  }
+
+  @Override
+  public void validate(FormatContext context) {
+    Schema schema = super.getSchema(context);
+    FailureCollector collector = context.getFailureCollector();
+    // When the sheet is specified by number, the sheet value must be a number
+    if (!conf.containsMacro(XlsInputFormatConfig.NAME_SHEET_VALUE)
+            && conf.getSheet().equals(XlsInputFormatConfig.SHEET_NUMBER)
+                    && (conf.getSheetValue() == null || !conf.getSheetValue().matches("[0-9]+"))) {
```

Comment:
shouldn't be checking numbers using regexes, just try to parse it into an int.

That being said, why are we overloading this single property to be either a name or a number? Seems simpler to have separate properties, one for sheet name and one for sheet number. Would be easier to describe what they each are and validate properly.

Reply:
The int validation is updated, now using parse int.

This behavior of having one drop down and overloading the other field is taken from the existing Excel source plugin.